src/main/java/com/coyoapp/tinytask/service/DefaultFileService.java

```diff
@@ -0,0 +1,60 @@
+package com.coyoapp.tinytask.service;
+
+import com.coyoapp.tinytask.exception.FileException;
+import com.coyoapp.tinytask.repository.FileRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Optional;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class DefaultFileService implements FileService {
+
+  @Autowired
+  private FileRepository fileRepository;
+
+  @PostConstruct
+  private void initImageRootFolder() {
+    Path path = fileRepository.initFilesDir()
+      .orElseThrow(() -> new RuntimeException("File base directory not accesssible."));
+    log.debug("initImageRootFolder(imageBaseDir={})", path.toAbsolutePath());
+  }
+
```

Comment:
due to comment above, method can be simplified by ...

```
  public String saveFile(MultipartFile file) {
    String fileName = UUID.randomUUID().toString();
    try {
      fileRepository.saveFile(fileName, file.getBytes());
    } catch (Exception e) {
      throw new FileException("File saving failed.");
    }
    return fileName;
  }
```

Reply:
"File reading failed." is wrong should be "File reading input MultipartFile failed.", `multipartFile.getBytes()` can throw IOException. 
`fileRepository.saveFile` can throw a bulk of different exceptions, which are simplified to single optional to make raise FileException fluent one-liner.